### PR TITLE
Fix SM100 D192 E5M2 attention sink accuracy

### PR DESCRIPTION
**Pull request by adshen**
## Before submitting

- [x] I agree to license this contribution under the terms of [LICENSE.txt](https://github.com/NVIDIA/cudnn-frontend/blob/develop/LICENSE.txt).
- [x] I ran `pre-commit run` and committed any formatting changes.
- [x] I added GitHub labels: one `cat-*`, one or more `mod-*`, and one `orig-*` (see [label list](https://github.com/NVIDIA/cudnn-frontend/labels)).

## Affected area

FE OSS kernels or CuTeDSL

## Summary

Fix the SM100 per-tensor FP8 `D_QK=192, D_V=128` forward kernel for E5M2 Q/K/V with an FP32 attention sink.

The change:

- removes the E5M2 sink routing restriction;
- initializes the E5M2 online-softmax state with the sink as a virtual score;
- uses true negative infinity for masked E5M2 sink scores;
- keeps the first real KV tile on a numerically stable max trajectory for BF16/FP16 output;
- consumes the combined sink/KV max and sum directly in the correction epilogue;
- converts the existing dtype rescale threshold to the kernel's log2 score units inside the kernel; and
- releases next-tile TMEM ownership only after the current output accumulator loads complete.

The review follow-up removes all `CUDNN_RESCALE_THRESHOLD` environment parsing, template-parameter plumbing, and cache-key specialization. E4M3 sink, no-sink, MXFP8, and other head-dimension routes retain their existing compile-time paths.

## Why

The FP8 reference initializes online softmax with `max=sink` and `sum=1`. The previous D192 kernel processed and quantized QK probabilities first, then merged the sink into the denominator in the final correction epilogue. Although equivalent in exact arithmetic, the two evaluation orders can choose different lazy-rescale maxima and therefore produce different E5M2 probability rounding near rescale cutoffs.

The sparse CI blocker had two mismatched output elements out of 17,372,160, with a maximum absolute error of `0.1484375` against an allowed absolute tolerance of `0.08`. Initializing the kernel state with the sink, preserving masked scores as true negative infinity, and using the correct threshold units reduces the same case's maximum absolute error to `0.03125` with zero mismatches.

The implementation is intentionally scoped to E5M2 with a sink. An ablation that removed the first-real-KV max update reproduced the FP16-output blocker with two mismatches and `max_abs=0.111145` (`atol=0.08`), while the FP8-output case still passed. The update therefore remains compile-time restricted to E5M2 sink with BF16/FP16 output; it does not change E4M3, no-sink, or FP8-output instruction paths.

The kernel maintains the following online-softmax invariant throughout the KV loop:

```text
sum = exp(sink - max) + sum_i exp(score_i - max)
O   = sum_i quantized_exp(score_i - max) * V_i / sum
```

## Related issues

None.

## API and compatibility impact

- No new public Python or C++ API.
- Enables the previously restricted SM100 D192/D128 per-tensor E5M2 + FP32 sink route.
- Does not read or introduce a rescale-threshold environment variable.
- Existing E4M3, no-sink, MXFP8, and sibling exact-shape routes are unchanged.
- No test tolerance, skip, or reference implementation was modified.

## Testing

Environment: SM100, cuDNN backend 9.26 (`92600`), PyTorch 2.11.0 + CUDA 13.0.

Formatting check:

```bash
black --check --line-length 160 --fast \
  python/cudnn/sdpa/fwd/engines.py \
  python/cudnn/sdpa/fwd/kernels/prefill_d192_d128_fp8_sm100.py \
  test/python/sdpa/frost/test_sdpa_graph_analyzer.py
```

Result: `3 files would be left unchanged`.

Graph analyzer:

```bash
pytest -q test/python/sdpa/frost/test_sdpa_graph_analyzer.py
```

Result: `84 passed`.

FROST FP8 coverage:

```bash
pytest -q test/python/sdpa/frost/test_sdpa_fwd_fp8_sm100.py
```

Result: `45 passed`. This covers masks, output dtypes, D192/D128, sink, GQA, padding, LSE, and scale contracts.

Random FP8 forward L0:

```bash
pytest -q test/python/test_mhas_v2.py::test_sdpa_fp8_fwd_L0
```

Result: `165 passed, 91 skipped`; 52 cases routed to `sdpa_fwd_prefill_sm100_d192_d128_fp8`.

The original focused failures (`test93`, `test110`, and `test145`) also passed repeatedly with fresh JIT caches. The final `test110` tensor capture reported zero tolerance mismatches and `max_abs=0.03125`.

## Performance impact

Measurements used serialized execution on the same SM100 GPU. Workload: `B=2`, `Hq=Hkv=128`, `Sq=Skv=8192`, `D_QK=192`, `D_V=128`, top-left causal mask, E5M2 Q/K/V with FP32 sink, 10 warmup iterations, and 40 timed iterations. The observed SM clock was 1860 MHz.

The initial online-sink implementation versus the previous late-merge implementation was within run-to-run noise:

| Output | Late-merge baseline | Online-sink implementation | Paired median delta |
|---|---:|---:|---:|
| BF16 | 5.364064 ms | 5.362096 ms | -0.086% |
| E4M3 | 5.402672 ms | 5.407648 ms | +0.092% |

After the review cleanup, three additional paired rounds compared the previous PR head with the final implementation:

| Output | Previous PR head | Final implementation | Delta |
|---|---:|---:|---:|
| BF16 | 5.363968 ms | 5.374320 ms | +0.193% |

The per-round deltas were `+0.002%`, `+0.265%`, and `+0.444%`. The final performance impact remains below 0.5%.

## Summary by CodeRabbit

- **New Features**
  - Added support for FP8 E5M2 inputs with sink tokens in the D192 attention path.
  - Improved softmax handling for sink configurations, including correct scaling, masking, and initial KV-tile processing.

- **Bug Fixes**
  - Removed an unnecessary restriction that prevented supported FP8 sink configurations from running.

- **Tests**
  - Updated coverage to verify FP8 E5M2 sink support.

### CODE REVIEW

**Review by vedaanta — CHANGES_REQUESTED** [review]
- can we move rescale_threshold * inv_log2 inside the kernel?
- lets remove CUDNN_RESCALE_THRESHOLD. seems like claude is replicating this from backend which is not needed.

**Comment by adshen** [comment]
@cudnn-ci-bot run frost

**Review by vedaanta — APPROVED** [review]
